staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -165,7 +165,7 @@ var (
 			Buckets:        []float64{0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1},
 			StabilityLevel: compbasemetrics.ALPHA,
 		},
-		[]string{requestKind},
+		[]string{LabelNamePhase, requestKind},
```

Comment:
@dgrisonnet - what are we doing when adding a label to an alpha metric?
Just a release note? or sth else?

Reply:
I do not understand this comment.  I do not understand why it is addressed to @dgrisonnet .

This PR is not changing what appears in a scrape.  This PR is changing the software structure to stop using the pair convenience layer for this Vec, because the members are not accessed in the usual paired way (i.e., a pair of numbers of waiting and executing requests).  The tightest member pairing for this Vec is a pair of mutating and readonly.

Reply:
The _pair_ Generator (now Vec, since #110101 merged) constructor puts under the pair covers an implicit label for the phase.  This PR removes the use of that pairing label so the phase label is no longer implicit.

Reply:
The actual change of this PR is visible if you ask GitHub to show you just the last commit.

Reply:
I have rebased, hopefully it is clearer now.

Reply:
Yes, there was/is.  Before this PR, line 151 invoked `NewSampleAndWaterMarkHistogramsPairVec` --- which adds the phase label to the given ones before proceeding down the stack; you can see that later in the display of this PR at line 48 in `sample_and_watermark.go`.  This PR changes line 151 here to invoke `NewSampleAndWaterMarkHistogramsVec` --- bypassing the pairing layer and explicitly passing the phase label from here.

Reply:
> what are we doing when adding a label to an alpha metric?

We should add an entry to the release note when that is the case since it is a user-facing change. Generally, since we don't have good discoverability mechanisms for metrics yet, I think it is a good practice to mention them in the release notes.

But yeah as Mike mentioned this is not a direct change to the metric, it already had the `phase` label before, it was just injected implicitly here: https://github.com/kubernetes/apiserver/blob/v0.23.1/pkg/util/flowcontrol/metrics/sample_and_watermark.go#L48.

Reply:
And I am proposing release notes in the relevant PRs; the switch from sample-and-watermark histograms to timing ratio histograms is user-visible.

Reply:
OK - I got it now - we're adding a new label here, but it was there already.

Is prometheus is smart and is doing deduplication if two labels have the same name?

Reply:
No dedup needed.  See other comments regarding the confusion between the code paths for two different vectors.

Reply:
OK - I finally got this. The previous function was actually adding that labels itself:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/sample_and_watermark.go#L48

Thanks for bothering with me.